File: Java/src/models/Processo.java
package models;

public class Processo implements Comparable<Processo> {

    private String name;
    private int time;
    private boolean interrupted;
    private int timeSpent;

    public Processo() {
    }

    public Processo(String name, int time, boolean interrupted, int timeSpent) {
        this.name = name;
        this.time = time;
        this.interrupted = interrupted;
        this.timeSpent = timeSpent;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getTime() {
        return time;
    }

    public void setTime(int time) {
        this.time = time;
    }

    public void setInterrupted(boolean interrupted) {
        this.interrupted = interrupted;
    }

    public boolean getInterrupted() {
        return interrupted;
    }

    public int getTimeSpent() {
        return timeSpent;
    }

    public void setTimeSpent(int timeSpent) {
        this.timeSpent = timeSpent;
    }

    @Override
    public int compareTo(Processo outroProcesso) {
        if (this.time > outroProcesso.getTime()) {
            return 1;
        } else if (this.time < outroProcesso.getTime()) {
            return -1;
        }
        return 0;
    }
}

File: Dart/escalonador/fcfs/models/processo.dart
class Processo implements Comparable<Processo> {
  String name = "";
  int time = 0;
  bool interrupted = false;
  int timeSpent = 0;

  Processo(
    this.name,
    this.time,
    this.interrupted,
    this.timeSpent
  );

  String get getName {
    return name;
  }

  set setName(String name) {
    setName = name;
  }

  int get getTime {
    return time;
  }

  set setTime(int time) {
    setTime = time;
  }

  bool get getInterrupted {
    return interrupted;
  }

  set setInterrupted(bool interrupted) {
    setInterrupted = interrupted;
  }

  int get getTimeSpent {
    return timeSpent;
  }

  set setTimeSpent(int timeSpent) {
    setTimeSpent = timeSpent;
  }

  int compareTo(Processo outroProcess) {
    if (this.time > outroProcess.getTime) {
      return 1;
    } else if (this.time < outroProcess.getTime) {
      return -1;
    }
    return 0;
  }
}
